packages/conversation/tests/rest/v1/conversation-domain-api.test.ts

```diff
@@ -0,0 +1,48 @@
+import { ConversationApi, Region, UnifiedCredentials } from '../../../src';
+
+describe('Conversation API', () => {
+  let conversationApi: ConversationApi;
+  let params: UnifiedCredentials;
+
+  beforeEach(() => {
+    params = {
+      projectId: 'PROJECT_ID',
+      keyId: 'KEY_ID',
+      keySecret: 'KEY_SECRET',
+    };
+  });
+
+  it('should initialize the client with the default "us" region', () => {
+    conversationApi = new ConversationApi(params);
+    conversationApi.getSinchClient();
+    expect(conversationApi.client).toBeDefined();
+    expect(conversationApi.client?.apiClientOptions.basePath).toBe('https://us.conversation.api.sinch.com');
+  });
+
+  it('should change the URL when specifying a different region', () => {
+    params.region = Region.EUROPE;
+    conversationApi = new ConversationApi(params);
+    conversationApi.getSinchClient();
+    expect(conversationApi.client?.apiClientOptions.basePath).toBe('https://eu.conversation.api.sinch.com');
+  });
+
+  it('should log a warning when using an unsupported region', async () => {
+    params.region = Region.CANADA;
```

Comment:
Should it be possible at all? I mean why Regions is generic - when you can provide conversationRegion param - I even see below the dedicated region enum for Conversation

Reply:
Javascript allows it all